core/variant/variant_op.h

```diff
@@ -824,6 +824,11 @@ class OperatorEvaluatorAddArray {
 	_FORCE_INLINE_ static void _add_arrays(Array &sum, const Array &array_a, const Array &array_b) {
 		int asize = array_a.size();
 		int bsize = array_b.size();
+
+		if (array_a.is_typed() && array_a.is_same_typed(array_b)) {
```

Comment:
Do we need to check `array_b.is_typed()`?
Apologies for the drive by comment... :^) Testing a github action PR review bot and was using your PR as an example.

Reply:
Technically I don't even think `array_a.is_typed()` needs to be checked. `is_same_typed` might just be enough because if they are both untyped then `set_typed` pretty much acts like a no-op. I guess it can be a bit expensive to call `set_typed` just to set no type though so it helps optimize that use-case.

To answer your question, checking if `array_b.is_typed()` would be redundant. If `array_a.is_typed()` and `array_a.is_same_typed(array_b)` then it can be safe to assume `array_b.is_typed()`.